uint/src/uint.rs

```diff
@@ -813,6 +813,16 @@ macro_rules! construct_uint {
 				arr
 			}
 		}
+
+		impl From<[u8; $n_words * 8]> for $name {
+			fn from(bytes: [u8; $n_words * 8]) -> Self {
+				let mut bytes = bytes;
+				bytes[..].reverse();
```

Comment:
but why we should reverse bytes? I guess, user expects that this conversion will just reinterpret  the bytes as `uint` without any additional transformations, am I wrong..? Or the reason behind it is that it just conflicts with former implementation?..

Reply:
> the reason behind it is that it just conflicts with former implementation?.

Yeah. I have to be honest: I expected the first impl to be correct, but the tests failed miserably so I banged on it in a bunch of different ways until they passed... :/ So I *think* the reason is this: the U* types are defined as `[u64; $n_words]` and index 0 is the lower end of the range (see sample below) *but* within the `u64` the number is little endian (on x86), so the last byte is the smallest.

The problem with my explanation is that if I'm right, then it should be enough to reverse the array *after* transmuting it. So I must be missing something! :(

```rust
		impl From<u64> for $name {
			fn from(value: u64) -> $name {
				let mut ret = [0; $n_words];
				ret[0] = value;
				$name(ret)
			}
		}
```

Reply:
U256 itself is little-endian

but default implementation from slice interprets _slice_ as big endian integer

i think you should use byteorder crate instead of unsafe transmute

Reply:
@NikVolf https://github.com/paritytech/parity-common/blob/6d47b026897bc940f97b1cb8ea56d79c3facda2a/uint/src/uint.rs#L777